backends/ceed-backend-list.c

```diff
@@ -0,0 +1,331 @@
+// Copyright (c) 2017-2025, Lawrence Livermore National Security, LLC and other CEED contributors.
```

Comment:
Note - part of the thinking here was that this would be one file so the list always matches the current compilation env. I was also thinking that since we add backends so infrequently this would help keep everything in one spot for easy reference.